### PR TITLE
Update (rewrite) the javaagent structure doc; document bootstrap modules

### LINKED ISSUE #4286: Document class loader implications of sharing instrumentation classes and multiple library dependencies
This is a tricky area that has come up a couple of times recently. I think it would be helpful to document somewhere to make sure we have a shared understanding of the issue(s) and solution(s).

e.g.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4236#issuecomment-933010972: 

> I think that it might be better to put KafkaClientsConsumerProcessTracing into a bootstrap module as when kafka-clients and kafka-streams are in different class loaders they would each get their own copy of that class. The same applies to KafkaConsumerIterableWrapper and KafkaConsumerIteratorWrapper that are used from other instrumentations, probably should have a KafkaWrapper interface in bootstrap module.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3718#discussion_r683246855:

> My understanding is that doesn't really work when 2 modules are in different class loaders. Muzzle will fail on spring-web because instrumentation uses classes from spring-webmvc hence they can't be in the same InstrumentationModule.

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Fixes #4286

### CODE REVIEW

File: docs/contributing/javaagent-jar-components.md [hunk 1]
```diff
@@ -1,102 +0,0 @@
...
-- `io/opentelemetry/javaagent/slf4j/` - contains SLF4J and its simple logger implementation, shaded
-during creation of `javaagent` jar file by Shadow Gradle plugin
-
-Available in the agent class loader:
-- `inst/` - contains `javaagent-tooling` and `javaagent-extension-api` modules and
-  `instrumentation` submodules, loaded and isolated inside `AgentClassLoader`.
-  Includes the OpenTelemetry SDK.
-
-![Agent initialization sequence](initialization-sequence.svg)
-[Image source](https://docs.google.com/drawings/d/1GHAcJ8AOaf_v2Ip82cQD9dN0mtvSk2C1B11KfwV2U8o)
```
Comment (mateuszrzeszutek) [inline docs/contributing/javaagent-jar-components.md:100]: I removed the initialization sequence doc from the new version of this page: it was completely out of date, and I don't think it was that useful -- the overall structure/architecture of the javaagent is very useful for a contributor (I think? it's not like I have anything to back this up 😅 ), but the initialization sequence is something that is changed very rarely. If you really need to know you can just read the code, it's not that hard to follow (unlike the structure, where you need to learn what every top-level module does, and which CL it's loaded by)

**Review by trask — APPROVED** [review]
nice ❤️
